src/main/java/modbus/IpConnection.java

```diff
@@ -0,0 +1,200 @@
+		case TCP:
+			params = new IpParameters();
+			params.setHost(host);
+			params.setPort(port);
+			master = new ModbusFactory().createTcpMaster(params, true);
+			break;
+		case UDP:
+			params = new IpParameters();
+			params.setHost(host);
+			params.setPort(port);
+			master = new ModbusFactory().createUdpMaster(params);
+			break;
+		default:
+			return null;
+		}
+
+		setMasterAttributes();
+
+		try {
+			master.init();
+			LOGGER.debug("Trying to connect");
+		} catch (ModbusInitException e) {
+			LOGGER.error("error in initializing master:" + e.getMessage() + " on " + host + ":" + port);
+			statnode.setValue(new Value("Could not establish connection"));
+			node.removeChild("stop");
+			makeStartAction();
+			try {
+				master.destroy();
+				LOGGER.debug("Close connection");
+			} catch (Exception e1) {
+				LOGGER.debug(e1.getMessage());
+			}
+		}
+
+		if (master.isInitialized()) {
+			link.masters.add(master);
+			return master;
+		} else {
+			return null;
+		}
+	}
+
+	String getAddDeviceActionName() {
+		return ADD_IP_DEVICE_ACTION;
+	}
+
+	@Override
+	Handler<ActionResult> getAddDeviceActionHandler() {
+		return new AddDeviceHandler(this);
+	}
+
+	@Override
+	void duplicate(ModbusLink link, Node node) {
+		// TBD
+	}
+
+	@Override
+	void removeChild() {
+		node.removeChild(ADD_IP_DEVICE_ACTION);
+	}
+
+	@Override
+	Action getEditAction() {
+		Action act = new Action(Permission.READ, new EditHandler());
+		act.addParameter(new Parameter(ATTR_NAME, ValueType.STRING));
+		act.addParameter(new Parameter(ATTR_TRANSPORT_TYPE, ValueType.makeEnum(Util.enumNames(IpTransportType.class))));
+		act.addParameter(new Parameter(ATTR_HOST, ValueType.STRING, new Value("")));
+		act.addParameter(new Parameter(ATTR_PORT, ValueType.NUMBER, new Value(502)));
+
+		act.addParameter(new Parameter(ATTR_TIMEOUT, ValueType.NUMBER, new Value(500)));
+		act.addParameter(new Parameter(ATTR_RETRIES, ValueType.NUMBER, new Value(2)));
+		act.addParameter(new Parameter(ATTR_MAX_READ_BIT_COUNT, ValueType.NUMBER, new Value(2000)));
+		act.addParameter(new Parameter(ATTR_MAX_READ_READ_REGISYER_COUNT, ValueType.NUMBER, new Value(125)));
+		act.addParameter(new Parameter(ATTR_MAX_WRITE_REGISTER_COUNT, ValueType.NUMBER, new Value(120)));
+		act.addParameter(new Parameter(ATTR_DISCARD_DATA_DELAY, ValueType.NUMBER, new Value(0)));
+		act.addParameter(new Parameter(ATTR_USE_MULTIPLE_WRITE_COMMAND_ONLY, ValueType.BOOL, new Value(false)));
+
+		return act;
+	}
+
+	private class EditHandler implements Handler<ActionResult> {
+
+		public void handle(ActionResult event) {
+			IpTransportType transtype;
+			try {
+				transtype = IpTransportType
+						.valueOf(event.getParameter("transport type", ValueType.STRING).getString().toUpperCase());
+			} catch (Exception e) {
+				LOGGER.error("invalid transport type");
+				LOGGER.debug("error: ", e);
+				return;
+			}
+			node.setAttribute("transport type", new Value(transtype.toString()));
+
+			readMasterParameters(event);
```

Comment:
Same comment here, you should name it clearer if there's a side effect. Here I dont really know what it does, nor do I see what object it acts upon.